### DISCUSSION TITLE
5-2. 지금 있는 곳에서 내가 가져가야 할 것

### DISCUSSION THREAD

**Discussion by imakerjun** [ssac-dev/yozm#16] [5장 다른 무대에도 서보고 싶어]
# 5-2. 지금 있는 곳에서 내가 가져가야 할 것

## 1. 동료와의 관계를 발전시킬 작은 도전 한 가지 시도하기
팀원과 갈등이 생겨 이직을 생각하는 경우가 있습니다. 물론 나와 맞지 않고 갈등이 반복된다면 함께 있는 것보다 떠나는 것이 더 나을 수도 있습니다. 단 갈등의 원인이 무엇이었는지, 그 갈등을 해결하기 위해 무엇을 시도했는지 생각해봐야 합니다. 누구나 회사에서 갈등을 겪습니다. 하지만 누군가는 업무에 큰 지장을 받지 않는 반면 누군가는 그것 때문에 괴로워하는 경우가 있습니다. 즉, 누군가는 이를 현명하게 해결한다는 것입니다. 갈등을 현명하게 해결하는 것은 직급이 올라갈수록 중요한 역량입니다.

새로운 회사에서도 인간관계에 관한 갈등이 없을 것이라는 보장이 없습니다. 그래서 갈등 상황 때문에 이직하고 싶은 마음이라면 할 수 있는 시도를 모두 해보는 것이 중요합니다. 최대한 시도해보고 그래도 갈등이 해소되지 않는다면 그때 이직을 준비하세요.

## 2. 기술적인 성장을 위한 작은 도전 한 가지 시도하기
기술적 성장에 대해 어떤 이미지가 떠오르나요? 팀을 이끄는 '기술 리더'가 되는 것인가요, 아니면 사회나 사용자에게 긍정적인 영향을 주는 프로젝트를 성공시키는 것인가요? 사실, 기술적 성장은 이보다 더 다양한 형태로 나타날 수 있습니다. 만약 '사회나 사용자에게 긍정적인 영향을 주는 프로젝트'가 기술적 성장에 대한 목표라면, 그러한 프로젝트를 구현할 수 있는 환경이 현재 주어져 있는지 자문해 보아야 합니다. 반대로, '조직 내에서 더 큰 책임과 영향력을 가지는 것'이 기술적 성장의 기준이라면, 현재 속한 조직이 그러한 역할을 수행할 수 있는 조건이 충족되어 있는지 검토해야 합니다. 더 나아가, 영향력을 높일 수 있는 동료나 멘토와 어떻게 협력할 수 있을지 계획을 세워야 합니다. 두 경우 모두에 있어, 기술적 성장은 단순한 스킬 향상을 넘어, 내가 추구하는 가치와 깊게 연결되어 있습니다. 이를 실현하기 위한 다양한 길이 있다는 것을 알아두는 것이 중요합니다.

## 3. 다른 사람에게 도전 거리를 공유하고 피드백 받기
내가 도전해보고 싶은 것, 그리도 도전 하고 있는 과정을 공유하고 어떻게 더 똑똑하게 해낼 수 있을지 피드백을 받아보세요. 분명 먼저 경험한 사람들의 지혜와 통찰을 얻을 수 있는 기회가 생각보다 많을 것입니다.

**Comment by kiljw316** [comment]
# 5-2. 지금 있는 곳에서 내가 가져가야 할 것

## 1. 동료 관계에서 작은 도전 한 가지

### 동료와의 생각을 공유하는 대화하기

회사에서는 같이 일하는 동료와의 의견 불일치로 인한 갈등이 발생할 수 있습니다. 이때 대화를 통해 서로의 의견에 대해서 목적이나 의도를 설명해 주면서 문제를 해결하기 위함이었다는 것을 전달합니다. 그리고 서로의 의견 중에 문제를 해결하는 데 더 도움이 되는 것을 고르기로 합니다. 그 과정에서 의견 차이가 좁혀지지 않을 수도 있기 때문에 또 다른 팀원에게 부탁해서 의견 차이를 좁히기 위한 도움을 받습니다.

## 2. 기술적인 성장을 위한 작은 도전 한 가지

### 나의 기술적 성장의 목표
- 다양한 문제를 해결하면서 기술적 역량을 높이는 것
- 회사에서만 얻을 수 있는 것을 경험해 보는 것

### 목표 달성을 위한 작은 도전
- `회사에서는` 문제해결 과정에서 어떤 결정을 내릴 때 이유 없이 내리지 말고 의식적으로 근거 있는 의사 결정을 한다.
- `회사 외적으로는` 개인적인 학습이나 토이 프로젝트로 기초 지식과 기술 학습을 해볼 수 있다.

**Reply by songsimo** [reply]
'동료와의 생각 공유'가 정말 중요하다고 생각해요. 요즘 일을 하며 느끼는게 이러한 소프트 스킬이 매우 중요하다는걸 많이 느끼고 있어요,  저는 갈등의 이유 중에 하나로 서로 믿지 못하기 때문이다를 꼽는데요. 서로 신뢰 관계를 형성하기 위해서는 이러한 생각 공유도 중요하다고 생각해요. 

kiljw316님은 다른 동료에게 어떤 부분을 부탁하여 의견 차이를 좁히는 좋은 방법이 있나요?

**Reply by kiljw316** [reply]
우선 댓글 감사드립니다. 🙇‍♂️ 

> kiljw316님은 다른 동료에게 어떤 부분을 부탁하여 의견 차이를 좁히는 좋은 방법이 있나요?

저는 여러 가지 레퍼런스들을 찾아와서 예시로 보여주는 방법을 주로 활용합니다. 예를 들어 어떤 주장에 대해서 내 의견대로 한다면 어떤 문제가 해결되는지를 설명해 주고 가능하다면 눈으로 볼 수 있게 간단한 예시도 작성해서 보여줍니다. 가장 중요한 건 동료에게 부담이 안 가는 선에서 의견을 제시합니다.

**Reply by hiyona-dev** [reply]
- **서로의 의견에 대해서 목적이나 의도를 설명**
- **레퍼런스들을 찾아와서 예시로 보여주는 방법**
정말 좋은 구체적인 방법 같습니다!
여러 경험을 통해서 생각을 공유하는 구체적인 의사소통이 필요하다는 것은 알겠으나, 구체적인 방법은 몰라서 구체적인 방법으로는 어떤 것이 있을까 막연했는데 글과 답변을 보면서 조금은 방법을 알 거 같습니다!

- **근거 있는 의사 결정**
좋은 자세 같아서 배울 점이네요! 저는 다행히 그동안 문제없이 옳은 결정을 해왔으나 막연하고 추상적이게 의사 결정을 해와서 이대로는 안 될 것 같아서 난감했는데요, 근거를 제시할 수 있는 의사 결정을 하면 되겠다는생각을 하게 되네요!

**Comment by lee5black** [comment]
# 5-2. 지금 있는 곳에서 내가 가져가야 할 것
---
이 부분에 대해서 정말 많이 고민을 해봤고, 아직도 답을 얻지 못했지만.
그럼에도 불구하고 생각나는것은
 
1. 서로 존중하며 일하는 문화
2. 개발 프로세스 (기획부터 배포까지)
3. 테스트 방법 및 이슈 분석법
4. 내가 어떤 사람인지 알기 (왜 이 환경에 만족 못하는지 뭘 더 갖고 싶은지 생각하기)

를 지금 있는 곳에서 가져갈 수 있을 것이라고 생각합니다

---
## 1. 동료 관계에서 작은 도전 한 가지
---
### 피드백 빨리하기

그동안 미숙한 것과 게을러서, 경력이 적다는 이유로 피드백을 느리게 드렸는데,
피드백이 느려지면 일의 텐션이 늘어지고 해야할 일을 제대로 못 끝내는 것 같습니다.

앞으로는 빠른 피드백을 위해 일 할때 더 집중하려고 합니다. 
또한 빠른 피드백을 위한 기술 습득을 게을리 하지 않을 것이고, 조그마한 개인시간을 투자하여 기술 습득 및, 업무에 필요한 반복 숙달, 불편사항 개선을 할 것 입니다.

일에 대한 지식이 늘고, 능력이 향상되어 업무 처리속도가 빨라지면 피드백이 빨라지고
같이 일하는 동료 관게에서도 긍정적인 영향을 줄 수 있을 것이라고 기대합니다.

---
## 2. 기술적인 성장을 위한 작은 도전 한 가지
---
요새 자기전 딱 5분만 C++ 서적 읽기를 하고 있습니다. (effective C++)

C++ 공부를 재밌게하려고 별의별 방법을 다 찾아봤는데 .. 찾지 못했습니다 (게임개발, qt 프로그래밍, 오픈소스 분석 및 기여, C++ 블로그 글 번역하기)

가지고 있는 실력에 비해 높은 목표를 지향 하느라 그런 것도 있고, 회사에서 쓰는 언어를 밖에서 쓰는게 쉽진 않았고
무엇보다 C++ 로 뭘 만들기가 힘들었습니다 (주제 잡기도 힘들고 ,뭔가 생각할게 되게 많았다고 생각했습니다)

다른 언어 (dart, JS)도 공부 해봤는데 하다가 그만둬서 요새는 딱 5분만 책을 읽고 있습니다.
(확실한 목표도 없고, 우선순위에 밀려 학습을 한번 놓치면 놔버리는 것 같습니다.)

5분이라는 시간이 생각보다 길고, 읽을 양도 많고 좋더라구요.
pdf 로 스캔을 떠서 자기전에 누워 아이패드 미니로 간단하게 읽는데 꽤 괜찮습니다

저는 출근길 15분 동안 밀리의 서재로 책을 읽는데 3년 읽으니 59권의 책을 읽었는데요
이때 꾸준함의 대단함에 대해서 알 수 있었습니다.
여기서 배운 것을 차용하여 자기전 5분 C++ 책 읽기를 고안하고 지금까지 잘 실천 중입니다 (3주 됐습니다)

제 도전에 대해 피드백 주시면 감사합니다 !

본인만의 재밌는 공부 방법도 소개해주시면 좋을 것 같아요 (다른 개발자 분들은 항상 어떻게 공부를 꾸준히 하시는거지? 궁금함이 있습니다 ㅠㅠ - 특히 C++... )

긴글 읽어주셔서 감사합니다 ~~

**Reply by kaori-killer** [reply]
피드백을 빨리하는 것을 저도 굉장히 중요하게 생각합니다. 하지만 저도 의지가 약해지는 부분이 문제가 되더라고요. 그래서 자동화 시스템을 통해서 일정 부분을 시스템에게 피드백을 맡겼더니 업무 효율이 올라갔습니다.

예를 들어, 첫 번째는 github actions 를 도입해서 코드에 대해서 lint 오류나 test 성공 여부를 지켰는지에 대한 피드백을 받을 수 있게 했습니다. 쉽게 놓칠 수 있는 부분을 잡을 수 있어서 좋더라고요. 두 번째는 모르는 부분을 공부할 수 있도록 만들었습니다. 내가 모르는 개념이 나올 때, 그 키워드를 메모장에 적어 놓으면, 하루가 끝날 때 지금까지 어떤 개념을 가장 많이 몰랐는지 나오는 시스템을 만들었습니다. 그러면 그 우선순위대로 개념을 잡으면 좋더라고요.

사소하지만 피드백을 빠르게 받을 수 있도록 도와주더라고요! 도움이 되셨으면 좋겠습니다 😄

---

저도 아이패드로 PDF 문서를 읽는 방법을 너무 좋아합니다. 출근 길에 59권이라니 대단하신데요? 👍 저도 자투리 시간을 활용해 5분만 더 공부하는 습관을 들이고 싶네요! 즐겁게 잘 읽었습니다 😄

**Reply by lee5black** [reply]
안녕하세요 ~ 댓글 남겨주셔서 감사합니다

자동화 시스템을 이용해서 피드백을 맡기는 아이디어 너무 좋은 것 같아요
특히 저는 "키워드를 메모장에 적어 놓으면, 하루가 끝날 때 지금까지 어떤 개념을 가장 많이 몰랐는지 나오는 시스템"
너무 괜찮다고 생각했습니다. 퇴근길에 한번 생각 하거나, 자기전에 어떻게든 한번은 찾아보고 잘거 같아요

기회가 된다면 어떤 시스템인지 구경 해볼 수 있을까요? :grinning:

**Comment by nara04040** [comment]
# 5-2. 지금 있는 곳에서 내가 가져가야 할 것

## 1. 동료 관계에서 작은 도전 한 가지 해보기

### 진행상황을 투명하게 공유하기
항상 미완성, 해결하지 못한 것을 공유하는 것을 부담스러워하여 그만큼 의사소통이 늘어짐을 느꼈었습니다. 이것을 해결하기 위해 미완성이라도 자신의 진행상황을 투명하게 공유합니다. 이에 따른 해결점을 같이 고민할 수 도 있으며, 진행상황을 알고있음으로 협업에 빠른 진행이 이뤄질 수 있다고 생각합니다. 
그러므로 저의 진행상황을 동료에게 투명하게 공유하여 빠른 피드백을 받을 것 입니다.

## 2. 기술적인 성장을 위한 작은 도전 한 가지 해보기

### 작은 기능을 자주 구현해보기
프론트엔드는 컴포넌트단위로 고민하고 구현하는 것이 많다고 생각합니다. 이런 생각을 기준으로 작은 기능을 자주 만들며 개념을 확장하는 방법을 시도하려 합니다!

## 3. 다른 사람에게 도전 거리를 공유하고 피드백 받기

**Reply by Seung-wan** [reply]
진행 상황을 투명하게 공유하는 것 정말 중요한 내용인 것 같아요!
원팀이 되기 위해서는 필요한 조건이고 팀으로써 문제를 해결해야 하기 때문에 만약 작업에 도움이 필요한 부분이 있다면 다른 동료가 도움을 줄 수도 있고 또 내가 다른 동료에게 도움을 줄 수도 있으니까요!

저도 이론 공부만큼이나 무언가를 만들어내는 것이 중요하다고 느껴요. 결국에 우리는 문제를 해결할 수 있는 무언가를 개발하는 사람이니까요!
내가 관심있는 도메인에서의 기능이라던가 혹은 디자인 시스템이라던가 작은것부터 하나하나 만들어보면 정말 좋을 것 같아요!

이 과정속에서 피드백을 받는 것도 서로 배워가는 것이 있기 때문에 의미있는 행동인 것 같습니다!

**Reply by sgsg9447** [reply]
작은기능을 자주 구현해보기는 개발자라면 정말 필요한것같아요.!
작은 성공을 계속해가다보면 어느새 성장해있는 모습을 볼 수 있을것같아요 
좋네요!

**Reply by Wu22e** [reply]
진행상황을 공유하는게 정말 중요하다고 생각해요!
데일리 스크럼을 통해서, 진행상황이 공유되지 않도록 방지하는것도 좋은 방법인 것 같아요

또, 한번에 큰 기능을 만들어낼 수 없듯이, 작은 기능에서 부터 설계가 시작된다고 생각합니다!

**Comment by sgsg9447** [comment]
## 1. 동료 관계에서 작은 도전 한 가지 해보기
- 질문 시 진행상황 공유하기
    - 질문시에는 현재 진행상황에 어떤 시도를 했는지를 케이스를 나눠 공유해보기
       - 인턴생활 중 진행 상황 중에서도 어떤 것에서 막혔는지 설명하는데 그부분이 시간이 오래걸리니 서로 답답했던 경험이 있습니다. 
       - 이런 부분을 잘 기록해두며 케이스를 나눠서 시각적으로 보여드리며 문제를 해결해야지라는 생각이 인턴생활이 끝나고서야 깨달았습니다.
       - 새로운 회사에 입사해서는 이부분을 꼭 적용해보고싶습니다.
## 2. 기술적인 성장을 위한 작은 도전 한 가지 해보기
- 새로운 회사 기술스택으로 아주 작은 사이드프로젝트 해보기
    - 새로운 회사의 합류를 2주정도 남긴 시점인데, Next, AntDesign, Firebase를 사용해보지않아서 이를 포함한 작은 사이드 프로젝트 해보기! 
    - 두려움을 미리 조금씩 채워나가기!
## 3. 다른 사람에게 도전 거리를 공유하고 피드백 받기
- 익스텐션 앱 개발 해야지… 하고 미뤄뒀는데! 다음주 돌아오기전까지 아주 작은 1단계 mvp라도 하고 오겠습니다!
     - 구글링 시 평가 이모지 표시하기
        → 구글링 검색 결과 화면 사이트 옆에 좋아요 직관적 표시
        → 방문한 페이지에서도 좋아요 볼 수 있음

**Reply by Seung-wan** [reply]
동료의 시간을 아껴주는 것도 정말 중요한 부분이라고 생각해요!
슬기님이 말씀해주신 것 처럼 정확히 어떤 시점에서 문제가 발생했는지, 어떤 방법들을 시도했었는지, 개발 환경(크롬, 사파리등)은 어떠한지 동료가 이해하기 위해 필요한 정보들을 잘 정리해서 전달해주신다면 양쪽 모두 좋을 것 같아요!

(저는 사실 입사전에 완전 노는걸 추천드리고 싶긴 한데...) 회사의 기술 스택으로 프로젝트 하는 것 너무 좋은 것 같아요!

개발된다면 저도 써보고 싶은 좋은 익스텐션이네요! 익스텐션 개발하는 것도 재밌으면서, 유용하기도 하고 좋은 것 같아요!

**Reply by nara04040** [reply]
혼자보다 함께 문제를 해결하면서 빠른 해결과 해결하는 방법에 대해서 진행상황을 공유한다는 부분이 공감됩니다. 저 또한 다른 팀플이나 회사에서 이점을 꼭 이행하고싶네요!!

항상 흥미롭게 생각했던 익스텐션 나중에 꼭 후기를 듣고싶네요! 😄

**Reply by Wu22e** [reply]
진행상황을 공유할때, 저는 개인적으로 고민 사항들 문서화했던 것들이 도움이 되었습니다! 기록을 해두면, 나중에 두번 일하지 않고 좋은 것 같아요

사이드 프로젝트를 통해서 회사의 스택을 더 깊게 이해할 수 있는 부분도 좋은 것 같습니다!
저는 개인적으로, 사이드 프로젝트를 통해 좀 다양한 기술들을 사용해보는 것을 선호하긴 하지만, 업무에 익숙한 기술들을 깊게 이해한 뒤에 해도 괜찮지 않을까 생각해봅니다!

**Comment by oeeen** [comment]
## 1. 동료 관계에서 작은 도전 한 가지 해보기

선, 후배를 가리지 않고 제가 알고 있는 내용을 잘 알려주려고 하고 있어요. 사소한 일이지만 저에게는 생각보다 도전적인 목표인데요. 이게 전달하는 말투나 방식에 따라서 약간 명령처럼 보이거나 좋지 않은 태도로 보이기도 하더라구요. 또한 받아주는 사람의 태도에 따라 반응이 다르기도 하구요. 무엇인가를 알려주려고 할 때는 그에 대한 근거나 결과가 확실해야한다고 생각하기 때문에 확실하게 알고 있다고 생각하는 것들도 다시 한 번 찾아보게 되면서 더 성장할 수 있는 것 같아요.

## 2. 기술적인 성장을 위한 작은 도전 한 가지 해보기

팀 내에서 자바 코드를 코틀린으로 전환하는 작업을 앞두고 있는데, 그것 때문은 아니었지만 코틀린 강의를 듣고 있어요. 처음 만들 때 클린 코드를 지향하며 만들어야 앞으로의 코드도 좋은 방향으로 가지 않을까 하는 생각으로 배운 내용들 정리해서 팀 내에 공유하는 자리를 가지려고 생각하고 있어요.

지금 회사 내에서 코틀린 스터디를 하고 있는데, 지금 책 스터디가 거의 끝나갈 때 쯤 강의에서 배운 내용들을 정리해서 공유하려고 해요.

**Comment by dq-hustlecoding** [comment]
1. 동료 관계에서 작은 도전 한 가지 해보기
회사일 말고, 인간적인 고민에 대해 공유하기 e.g. 인간관계 등

2. 기술적인 성장을 위한 작은 도전 한 가지 해보기
경험해보지 못한 기술을 이용해서 제품을 출시하기 - Flutter 로 모바일 앱 출시하기

**Reply by oeeen** [reply]
제 기준에서는 앱을 출시한다는 건 엄청 큰 도전이라 생각해서 작은 것부터 하면 어떨까 하는 생각인데 ㅎㅎㅎ 대단하시네요! 모바일 앱 출시하게 되면 공유한번 부탁드려요 ㅎㅎㅎ

**Comment by applebuddy** [comment]
### 1. 동료 관계에서 작은 도전 한 가지 해보기
- 팀원 한분과 최근 시작한 기술포스팅 스터디(매달 포스팅 인증 안하면 벌금, 자기 포스팅 소개하기) 꾸준히 진행하기

### 2. 기술적인 성장을 위한 작은 도전 한 가지 해보기
- 내 지식을 활용해서 전자책 작성하기

**Comment by lsw1164** [comment]
# 5-2. 지금 있는 곳에서 내가 가져가야 할 것

## 1. 동료 관계에서 작은 도전 한 가지 해보기

팀원들과 1on1을 통해, 피드백 구하기.

**피드백 구하기**, 평소에는 크게 생각하지 않고 넘어가기 쉬운 거 같습니다. 회사 평가 시즌에 억지로 시켜서 한 기억밖에 없네요. 그냥저냥 별 문제 없이 흘러가고 있으면 굳이 피드백을 구하지 않습니다. 일하다보면 약간 불편해도 그냥 좋은게 좋은거지 하면서 넘어가는 부분이 상당히 많습니다. 어쩌면 동료들도 나를 향한 불편함을 그냥 좋게 넘기지 않았을까 궁금해졌습니다. 동료 피드백이 없이 스스로를 돌아보는 것만으로는 알기 어렵다고 생각했습니다. 동료들이 생각하는 나의 강점과 개선하면 좋은 점들에 대해 대화해보겠습니다.

## 2. 기술적인 성장을 위한 작은 도전 한 가지 해보기

사내 인프라 스터디에서 동료들에게 공유할 거 한 가지 이상 준비해보기 하겠습니다.

**Reply by kiljw316** [reply]
팀원들의 피드백은 정말 스스로는 알지 못하는 것들을 알 수 있게 해준다는 점에서 필수인 것 같아요. 회사 평가 시즌에 급하게 요청하면 꼭 필요한 피드백을 해줄 시간이 없어서 좋은 피드백을 받기는 어려운 것 같습니다. 좋은 피드백을 받기 위해서라도 평소에 1 on 1 같은 방법으로 소통 한다면 더 좋을 것 같네요.

**Comment by songsimo** [comment]
1. 동료 관계에서 작은 도전 한 가지 해보기

* 대화 혹은 다른 방법을 통하여 동료와 신뢰 관계를 쌓기

일을 하다보니 의견이 다른 경우가 자주 발생했습니다. 처음에는 강하게 의견을 표출해보기도 하고  사전 조사를 하여 준비한 것으로 설득을 했지만 원하는 결과로 진행되지 않았습니다. 이런 일이 자주 생기면서 의견 충돌로 발생하는 에너지 소비보다는 상대방의 의견을 따르는 수동적인 자세로 변했습니다. 시간이 흘러 곰곰히 생각해보니 동료와 저의 사이에 신뢰관계를 쌓지 않은 상태로 너무 제 의견을 표출하지 않았나 생각이 들었습니다. 

2. 기술적인 성장을 위한 작은 도전 한 가지 해보기
* 동료에게 먼저 사내 스터디를 제안하고 다음 프로젝트에 적용하면 좋을 기술들을 학습하기
* 회사 외적으로 공부하고 있던 기술들을 적용하여 토이 프로젝트 진행하기

**Reply by kiljw316** [reply]
사실 동료와 신뢰 관계가 제일 중요하죠. 

저는 상대방 입장에서 생각해 보는 것과 신뢰 관계 구축을 동시에 하는걸 추천드립니다. 

말씀해 주신 것처럼 사전 조사를 하거나 강하게 의견을 표출하는데도 설득되지 않았다면 상대방 입장에서는 부담이 되었을 가능성이 있다고 생각해요. 그래서 기술적인 성장을 위한 도전에 작성해 주셨다시피 스터디를 하거나 자료만 먼저 공유 해서 상대방에게 @songsimo  님의 의견이 부담이 없다는 것을 인식시키고 이런 방식으로도 문제를 해결해 볼 수 있다는 것을 어필한다면 의견을 반영하시는 데 도움이 될 것으로 생각합니다.

**Reply by songsimo** [reply]
네! 맞아요😃 나중에 지나고 나서 보니 상대방 입장에서는 주어진 책무와 상황, 저와의 관계 등 여러가지 고려할게 있었어요. 지금은 이해하고 어떻게 해야할까 고민이었는데, 지금은 후임부터 신뢰 관계를 쌓으려고 노력하고 있어요. 말씀하신 것들을 후임과 해봐야겠어요, 감사합니다☺️.

**Comment by kaori-killer** [comment]
# 동료 관계에서 작은 도전 한 가지

- 스크럼 방식을 통해 팀 내부에 빠른 피드백이 돌 수 있게 만들기

저는 요즘 회사가 아닌 주변 사람들과 토이 프로젝트를 하고 있습니다. 실력을 떠나 팀장을 맡게 되었는데요. 처음 팀을 이끌다보니 문제가 많았습니다. 

같은 회사가 아니다 보니 시간도 장소도 맞추기 힘들었습니다. 그래서 데일리 스크럼도 불가능했고 서로의 힘든 점을 공유하기가 어려웠습니다.

그래서 고민을 하던 중 ‘애자일 개발이 처음인 내가 스크럼 마스터가 된 건에 관하여‘ 라는 책을 발견했습니다.

이 책은 ’스크럼‘ 을 손쉽게 팀에 도입하는 방법을 다루고 있습니다. 스크럼은 애자일하게 개발하는 방식 중 하나기도 합니다.

이 방식을 팀에 잘 도입해서, 서로의 상황이 빠르게 공유되고, 지속적인 피드백을 받을 수 있는 환경을 만들고 싶습니다.

# 기술적인 성장을 위한 작은 도전 한 가지

- Front-End 기초에 대한 공부, HTML부터 다시

저는 최근에 기초가 중요하다는 것을 다시금 느꼈습니다. REST API 에 대한 정확한 개념을 모르고 API 명세서를 만들고 있는 저를 보면서 말입니다. 그래서 틈이 나는대로 처음부터 볼 예정입니다. 일단 HTML MDN 문서부터 보려고 합니다. 생각해보면 의미있는 신택스 태그보다 Div로 박스를 처리할 때가 많았던 것 같아서요. 더 수준 높은 기술에 대해 공부하는 것도 좋지만, 지금은 기초 튼튼이 답인 것 같습니다. 😊

**Reply by lee5black** [reply]
안녕하세요 ! 작성해주신 도전 잘 읽어봤습니다

말씀주신 책 궁금해지네요 제목도 흥미롭습니다 ㅋㅋ

지금 경험하고 계신 팀장 역할이 앞으로 성장에 큰 자산이 될  것 같다는 느낌이 확 왔습니다
응원 할게요 !!

또 느낀점은 기술적인 성장은 위해 역시 기초를 탄탄히 해야하는군요 .. 저도 놓친건 없는지 한번 되돌아 봐야겠다고 생각했습니다

경험 공유 해주셔서 감사합니다~~


ps. 혹시 토이 프로젝트에 대해 간단하게 소개 해줄 수 있나요? 궁금합니다 :relaxed:

**Comment by hiyona-dev** [comment]
# 5-2. 지금 있는 곳에서 내가 가져가야 할 것

## 1. 동료 관계에서 작은 도전 한 가지 해보기

- 구체적으로 설명하고 솔직하게 대화하며 빠르게 피드백하기
  > 최근 건너 팀원의 급작스러운 퇴사를 하시고 난 뒤의 지인 회사 상황에 대한 이야기를 들었습니다. 실력도 있고 성격도 좋으신 분 같았는데 왜 그분이 나가고 난 뒤의 분위기는 좋지 않았나 싶었습니다. 그리고 자초지종을 듣고 나서 동료 관계에 대한 깨달음을 얻었습니다.
  > 긁어 부스럼을 만들고 싶지 않은 사람들과 구체적인 상황설명 및 공유를 하지 않는 팀원 간의 의사소통에서 비롯된 불만이 문제가 되었습니다. 서로 맞지 않는다는 생각을 심어주게 되어 개개인으로 봐서는 전부 좋은 사람들임에도 불구하고 업무적으로 엮여 좋지않게 헤어지는 케이스였습니다.
  > 긁어 부스럼을 만들고 싶지 않다는 생각으로 말을 참기보다는 대화를 부드럽게 시도하며 구체적으로 질문을 해야 함과 아는 것은 안다고 말하고 모르는 것은 모르겠다고 말하는 솔직함도 있어야 하며, 개인적인 부분으로 느껴져도 업무 차질을 발생시키지 않기 위한 자세한 설명도 필요함을 알게 되었습니다.
  >사람은 감정과 생각이 있기 때문에 어떠한 말을 해도 오해가 생길 수 있습니다. 말을 듣고 의미를 해석하는 것은 청자의 몫이기 때문입니다. 그래서 대화할 때 상대가 의미를 해석하게 하는 대화가 아닌, 오해도 곡해도 없는 구체적이고 꾸밈없는 솔직히 대화해야겠다고 생각하게 되는 경험이었습니다.
  > 지금 당장은 실력이 부족한 것은 당연한 상황이니 진도가 부진하더라도 피드백을 빨리 빨리하는 것이 중요하겠다는 생각도 하게 되었습니다. 내가 머뭇거리는 사이에 다른 팀원은 막연히 기다리기만 하는 상황이 연출될 테니까요!

## 2. 기술적인 성장을 위한 작은 도전 한 가지 해보기
- 팀 단위 토이 프로젝트 경험하기
  > 회사에서 팀 단위로 협업하게 될 텐데 단순 혼자 클론 코딩으로만 끝낼 것이 아니라 토이 프로젝트를 진행하면서 다른 사람과의 협업을 경험하고 싶습니다. 나는 실력이 부족한 것이 맞고, 창피하다고 뒤로 물러서기만 해서는 제자리가 될 것을 다시 깨닫습니다.

- 기술 서적 읽고 기록하기
  > 자기 전 기술 서적을 읽고 내용을 기록하여 머리에 남도록 해야겠습니다.
  > 자신감이 없는 것은 별개로 기술적인 성장을 위해서는 마주 보고 부딪히며 나아가야 함을 다시 상기합니다.

**Reply by kiljw316** [reply]
꽤 길고 구체적으로 작성해 주셨네요. 잘 읽었습니다.

> 구체적으로 설명하고 솔직하게 대화하며 빠르게 피드백하기

구체적으로 설명하고 솔직하게 대화하며 빠르게 피드백하는 것은 정말로 좋은 방법 같습니다. 
추가로 `상대방의 입장에서 생각해보기` 도 하신다면 좋을 것 같습니다. 
구체적으로 설명하는 것도 상대방이 상황에 대해 이해를 하기위해 정말 좋은 방법이지만 상대방 입장에서 구체적인 설명이 없어도 되는 경우도 있습니다. 그런 경우에는 구체적인 설명이 오히려 상대방에게 집중력이나 시간을 뺏을 수도 있기 때문에 상대방에게 자세한 설명을 해야 하는지 아니면 간단하고 빠른 소통이 필요한지를 먼저 파악한다면 좋을 것 같네요. 👍 

> 팀 단위 토이 프로젝트 경험하기
> 기술 서적 읽고 기록하기

프로젝트 경험과 지식 학습을 둘 다 해보는 것은 정말 좋은 것 같습니다. 먼저 목표를 설정하고 시간 분배나 학습의 우선순위를 정하고 나서 시작하신다면 더 효율적인 공부가 될 것 같습니다.

**Comment by qotjsry** [comment]
# 5-2. 지금 있는 곳에서 내가 가져가야 할 것
 ## 1. 동료 관계에서 작은 도전 한 가지 해보기
내가 아는 지식을 최대한 전파하고 학습에 대한 의욕을 만들어 주기 서로 개발에 대해 학습을 하고 내용을 공유 하면서 성장해야 된다고 생각합니다. 혼자서 드를 리팩토링하고 지식에 대해 학습을 한다면 재미가 떨어지고 습득 속도가 떨어질 수 밖에 없을거 같습니다. 그러면 양질의 코드가 나올 수 없기 때문에 다같이 성장하는 방향으로 이끄는걸 도전하고 있습니다.

 ## 2. 기술적인 성장을 위한 작은 도전 한 가지 해보기
 매일 일찍 출근하여 기초 서적 독서 매일 퇴근 후 알고리즘 문제 1~2문제 공부 기술적인 성장을 위해 기초부터 다시 차근차근 쌓아 올리려고 합니다.
